src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/InstanceData.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    ///     A holder of instance data.
+    /// </summary>    
+    public class InstanceData
+    {
+        private readonly string _instanceName;
+        private readonly CounterSample _sample;
+
+        public InstanceData(string instanceName, CounterSample sample)
+        {
+            _instanceName = instanceName;
+            _sample = sample;
+        }
+
+        public string InstanceName
+        {
+            get
+            {
+                return _instanceName;
+            }
+        }
+
+        public CounterSample Sample
+        {
+            get
+            {
+                return _sample;
+            }
+        }
+
+        public long RawValue
```

Comment:
`public long RawValue { get; }`

Reply:
How does the getter know that RawValue would come from Sample.RawValue?

Reply:
my mistake sorry. It's fine like that.